### PR TITLE
Improve Cupertino docs

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

Adds "See also" section to cupertino library
Adds 3 snippets for `CupertinoApp` (2 with screenshots require https://github.com/flutter/assets-for-api-docs/pull/129 to merge first)
Adds a dartpad for `CupertinoNavigationBar`
Adds a dartpad for `CupertinoPageScaffold`

## Related Issues

https://github.com/flutter/flutter/issues/72926

## Tests

No tests, only documentation.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/app.dart
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/app.dart:98]: nit:
```suggestion
/// will be used for Cupertino widgets in the app.
```

File: packages/flutter/lib/src/cupertino/nav_bar.dart
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:221]: nits:
```suggestion
///     child: Column(
///       children: [
///         Container(height: 50, color: CupertinoColors.systemRed),
///         Container(height: 50, color: CupertinoColors.systemGreen),
///         Container(height: 50, color: CupertinoColors.systemBlue),
///         Container(height: 50, color: CupertinoColors.systemYellow),
///       ],
///     ),
```

File: packages/flutter/lib/src/cupertino/page_scaffold.dart
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/page_scaffold.dart:24]: nit: Just one new line is enough here
```suggestion
```

File: packages/flutter/lib/src/cupertino/page_scaffold.dart
Comment (shihaohong) [inline packages/flutter/lib/src/cupertino/page_scaffold.dart:32]: This comment about the background color seems unrelated to the sample itself. Is this intended? If it is, add it to the dartpad description above

**Review by shihaohong — APPROVED** [review]
LGTM with nits and once the `assets-for-api-docs` change is merged.

File: packages/flutter/lib/src/cupertino/page_scaffold.dart
Comment (guidezpl) [reply packages/flutter/lib/src/cupertino/page_scaffold.dart:32]: Yes, modified

**Pull request merged**
